src/index.js

```diff
@@ -162,7 +162,8 @@ function injectReactDocgenInfo(className, path, state, code, t) {
       resolver = require('react-docgen').resolver[state.opts.resolver];
     }
 
-    docObj = reactDocs.parse(code, resolver);
+    const parsedDocs = reactDocs.parse(code, resolver);
+    docObj = Object.prototype.toString.call(parsedDocs) == '[object Array]' ? parsedDocs[0] : parsedDocs;
```

Comment:
I don't think the first array item will always be the correct docgen content to be returned. I think we need to create multiple docgen items that map to the correct components that are exported in the first place.

Reply:
Yeah, I was thinking about doing something like this, but didn't devote the time yet. In the current version, `babel-plugin-react-docgen` returns nothing when using any resolver but the default. I saw this more as a bug fix rather than a feature. I thought returning the first export at least would be an improvement over nothing.

Reply:
To clarify, this is when using a non-default resolver, like `findAllExportedComponentDefinitions` and there is more than one export. There should probably be a test case for this, too.

Reply:
Yeah, I think I understand what you're saying.

For example, if I have something like this:

```
export const Button = (text) => {
  return <button>{text}</button>;
}
Button.propTypes = {
  text: PropTypes.string
}

export default const ButtonWithLink = (text, href) => {
  return <a href={href}><Button>{text}</Button></a>;
}
ButtonWithLink.propTypes = {
  text: PropTypes.string,
  href: PropTypes.string
}
```

I think I'd expect both `ButtonWithLink.__docgenInfo` and `Button.__docgenInfo` to contain the appropriate info for each one.

If I'm understanding the `react-docgen` docs correctly, the array returned contains all the components in the file in order. This would mean that all exported components would get the first component's info. In this example, `ButtonWithLink` would wrongly be applied the `Button` info.

Don't worry too much about this, it's not your fault the code is terribly written. I think this plugin is long overdue for a major refactor but I haven't found time to do it yet.

Reply:
Yes, that is exactly how `react-docgen` returns the info. Each components' prop info is an object in an array with the default export's name. Like you said, any additional exports are erroneously contained in this default export's prop info. Because `addon-info` is expecting a single object for the prop info, it doesn't use the array that's provided by this babel plugin. (In addon-info/makeTableComponent.js).
`return obj && obj.props && (0, _keys2.default)(obj.props).length > 0;`

Even if it did, when using `findAllExportedComponentDefinitions` with more than one component in a file, some info still wouldn't be available under the correct component's name.

In order to fix this properly, I think we'd have to change both `addon-info` and this plugin. We'd need to output the info from this plugin under each component's name and `addon-info` would need to be altered to accept and use this new format.

My intent with this PR was to at least make the info from one component available to addon-info, as opposed to it not finding any info at all. Alternatively, we could submit a PR in `addon-info` to accept the array, but using the babel-plugin in it's current state, the info of the correct component wouldn't always be shown, anyway.

Reply:
From issues here and in `addon-info`, I see there are many people having trouble with info not showing up for their props (or descriptions and proptypes missing due to `addon-info` not finding the docgen info). I see commits [here](https://github.com/storybooks/babel-plugin-react-docgen/commit/708769980b03c8564a4646cc3434019b8416e644) that seem to intend for the other resolvers to be usable, but with more than one export, this plugin returns an array mislabeled as the default export—using anything but the default resolver is not going to work properly.